EIPS/eip-3085.md

```diff
@@ -48,16 +48,29 @@ The wallet **MAY** reject the request for any reason.
 `wallet_addEthereumChain` accepts a single object parameter, specified by the following TypeScript interface:
 
 ```typescript
-interface AddEthereumChainParameter {
+interface BaseNativeCurrency {
+  name: string;
+  symbol: string;
+}
+
+interface NativeCurrencyWithDecimals extends BaseNativeCurrency {
+  decimals: number;
+}
+
+interface NativeCurrencyWithScalingFactor extends BaseNativeCurrency {
+  scalingFactor: string;
+}
+
+type NativeCurrency =
+  | NativeCurrencyWithDecimals
+  | NativeCurrencyWithScalingFactor;
+
+interface AddEthereumChainParameter<T extends NativeCurrency> {
   chainId: string;
   blockExplorerUrls?: string[];
   chainName?: string;
   iconUrls?: string[];
-  nativeCurrency?: {
-    name: string;
-    symbol: string;
-    decimals: number;
-  };
+  nativeCurrency?: T;
```

Comment:
This doesn't need to be generic, and I suspect that unnecessarily complicates things.
```suggestion
nativeCurrency: NativeCurrency
```
(though, see my comment above about just switching to `scalingFactor` only)